src/proposals/Proposal.sol

```diff
@@ -27,78 +27,81 @@ abstract contract Proposal is Test, Script, IProposal {
     /// they all follow the same structure
     Action[] public actions;
 
-    /// @notice debug flag to print proposal actions, calldata, new addresses and changed addresses
-    /// @dev default is true
-    bool internal DEBUG = false;
+    /// @notice debug flag to print internal proposal logs
+    bool internal DEBUG;
+    bool internal DO_DEPLOY;
+    bool internal DO_AFTER_DEPLOY_MOCK;
+    bool internal DO_BUILD;
+    bool internal DO_SIMULATE;
+    bool internal DO_VALIDATE;
+    bool internal DO_PRINT;
 
     /// @notice Addresses contract
     Addresses public addresses;
 
-    /// @notice the actions caller name in the Addresses JSON
-    string public caller;
-
     /// @notice primary fork id
     uint256 public primaryForkId;
 
-    modifier buildModifier() {
-        _startBuild();
+    /// @notice buildModifier to be used by the build function to populate the
+    /// actions array
+    /// @param toPrank the address that will be used as the caller for the
+    /// actions, e.g. multisig address, timelock address, etc.
+    modifier buildModifier(address toPrank) {
+        _startBuild(toPrank);
         _;
-        _endBuild();
+        _endBuild(toPrank);
```

Comment:
I thought about this problem and the conclusion I came to was that using the modifier is the best way, lmk if you think different:

 In order to get the address to prank, we would need to have it saved to storage and have a place to set it. This could be done in two ways:
1. The constructor could be used (as before), but then we would also need to create the Address object in the constructor to get the address to prank from Addresses storage. We can't create it the Address contract outside of the proposal and pass a reference if we want to enable running the forge script through the run function.
2. We could have a setter function for the address to prank, which would need to be called at some point in the inherited contracts. I believe the best place for this would be the `build` function, but then we have basically the same flow as we achieve using the modifier